infra/gravity/status.go

```diff
@@ -17,30 +17,141 @@ limitations under the License.
 package gravity
 
 import (
+	"bufio"
 	"context"
+	"encoding/json"
 
 	"github.com/cenkalti/backoff"
 	"golang.org/x/sync/errgroup"
 
-	"github.com/gravitational/robotest/lib/constants"
 	sshutils "github.com/gravitational/robotest/lib/ssh"
 	"github.com/gravitational/robotest/lib/utils"
 	"github.com/gravitational/robotest/lib/wait"
 	"github.com/gravitational/trace"
 )
 
+// ClusterState* consts come from https://github.com/gravitational/gravity/blob/7.0.0/lib/ops/constants.go#L64-L93
+const (
+	// ClusterStateActive is healthy and not running any operations.
+	ClusterStateActive = "active"
+	// ClusterStateDegraded is unhealthy.
+	ClusterStateDegraded = "degraded"
+)
+
+// SystemStatus comes from https://github.com/gravitational/satellite/blob/7.1.0/agent/proto/agentpb/agent.pb.go#L28-L32
+type SystemStatus int
+
+const (
+	SystemStatus_Unknown  SystemStatus = 0
+	SystemStatus_Running  SystemStatus = 1
+	SystemStatus_Degraded SystemStatus = 2
+)
+
+var SystemStatusNames = [...]string{"undefined", "running", "degraded"}
+
+func (ss SystemStatus) String() string {
+	return SystemStatusNames[ss]
+}
+
+// UnmarshalJSON fulfills the Unmarshaler interface, allowing SystemStatus
+// to perform custom JSON unmarshalling.
+//
+// Gravity 5.2.x+ marshals SystemStatus as an integer and Gravity 5.0.36-
```

Comment:
I would prefix this part with `TODO` so this potentially can be searched for later.

Reply:
Added a TODO in 3051902.